clang/test/CIR/CodeGen/builtin-floating-point.c

```diff
@@ -0,0 +1,1177 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -ffast-math -fclangir-enable -emit-cir %s -o - | FileCheck %s
+
+// ceil
+
+float my_ceilf(float f) {
+  return __builtin_ceilf(f);
+  // CHECK:      cir.func @my_ceilf
+  // CHECK-NEXT: %0 = cir.alloca f32, cir.ptr <f32>, ["f", init] {alignment = 4 : i64}
+  // CHECK-NEXT: %1 = cir.alloca f32, cir.ptr <f32>, ["__retval"] {alignment = 4 : i64}
+  // CHECK-NEXT: cir.store %arg0, %0 : f32, cir.ptr <f32>
+  // CHECK-NEXT: %2 = cir.load %0 : cir.ptr <f32>, f32
+  // CHECK-NEXT: %3 = cir.ceil %2 : f32
+  // CHECK-NEXT: cir.store %3, %1 : f32, cir.ptr <f32>
+  // CHECK-NEXT: %4 = cir.load %1 : cir.ptr <f32>, f32
+  // CHECK-NEXT: cir.return %4 : f32
+}
```

Comment:
Could you reduce the tests to check only what has been added? E.g. the codegen of the builtin:

```c++
float my_ceilf(float f) {
  return __builtin_ceilf(f);
}
// CHECK: cir.func @my_ceilf
// CHECK:   %{{.+}} = cir.ceil %{{.+}} : f32
```

Reply:
IIUC that wouldn't catch mixing up two builtins though. e.g. if I accidentally generated `log2` for `__builtin_log10` and `log10` for `__builtin_log2`.

Reply:
I believe it would work without any issues (assuming the test is written correctly). For example, the test below should fail if `__builtin_log2` were to generate a `cir.log10` instead of a `cir.log2`:

```
double my_log2(double f) {
  return __builtin_log2(f);
  // CHECK: cir.func @my_log2
  // CHECK:   %{{.+}} = cir.log2 %{{.+}} : f64
}

double my_log10(double f) {
  return __builtin_log10(f);
  // CHECK: cir.func @my_log10
  // CHECK:   %{{.+}} = cir.log10 %{{.+}} : f64
}
```

The trailing types (e.g. `: f64`) should ensure the distinction between float and double builtins.

Do you have an example of a test like the one above that might return a false positive?

Reply:
Ah, ok. I was under the impression that `// CHECK:` ignored the order, but apparently it doesn't actually. Never mind then.